src/DGtal/geometry/curves/AlphaThickSegmentComputer.h

```diff
@@ -0,0 +1,711 @@
+// Inclusions
+#include <iostream>
+#include "DGtal/base/Common.h"
+#include "DGtal/base/ReverseIterator.h"
+#include "DGtal/kernel/SpaceND.h"
+#include "DGtal/geometry/surfaces/ParallelStrip.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+/////////////////////////////////////////////////////////////////////////////
+// class AlphaThickSegmentComputer
+/**
+ * Description of class 'AlphaThickSegmentComputer' <p>
+ *
+ * \brief Aim: This class is devoted to the recognition of the alpha
+ * thick segment as described in \cite FaureTangential2008 . From a
+ * maximal diagonal alphaMax thickness, it allows to apply the recognition
+ * of a thick segment with the ability to take into accounts some
+ * noise. Moreover the segment can be detected from points not
+ * necessary connected and/or with floating coordinates.
+ *
+ * As other solutions like \cite DebledRennessonBlurred2005 the
+ * algorithm given here is based on the height/width computation of
+ * the convex hull computed from a given set of points. The actual
+ * implementation exploits the height/width update defined from
+ * the \cite FaureTangential2008 (see \cite FaureTangential2008 page
+ * 363) which reduces the complexity from \f$O(n\ log\ n) \f$ into
+ * \f$O( log\ n) \f$.  Note that the convexhull update in linear
+ * time (with point substraction) proposed by Buzer \cite
+ * lilianComputing2007 is not yet implemented.
+ *
+ *
+ *
+ * @tparam TInputPoint the type of the input point which have to be of
+ * dimension 2.
+ * 
+ * @tparam TConstIterator the type of iterator of candidate points
+ * (used in the init with iterator) which should be readable and
+ * forward. By default the iterator is set to the const_iterator of std::vector< TInputPoint > .
+ *
+ *
+ * This class is a model of ForwardContainer and  CForwardSegmentComputer.  
+ * It is also default constructible, copy  constructible, assignable and equality comparable.
+ *
+ *
+ * A typical application of an alpha thick segment recognition can be
+ * done as follows:
+ * - If you consider input point with floating coordinates, you can define this type:
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementNoisyTypedef
+ *
+ * - Then import eventually a vector containing the input points by using the PointListReader class:
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementNoisyReadFile 
+ * 
+ * - Finally apply the segment recognition (here of maximal thickness 15)  by adding the sequence (forward) of contour points: 
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementNoisInitAndReco 
+ *  @note The maximal thickness given in initialization is the same than
+ *  the width given by the method width() of the ParallelStrip primitive
+ *  (not the vertical/horizontal width of the convex hull).
```

Comment:
Question: the thickness is not a vertical/horizontal width ? So the thickness is the true width of the strip ?